server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/table/ClientAddress.java

```diff
@@ -19,15 +18,25 @@
  * @author Pedro Ruivo
  * @since 9.2
  */
+@ProtoTypeId(ProtoStreamTypeIds.SERVER_HR_CLIENT_ADDRESS)
 public class ClientAddress implements Address {
 
-   public static final AdvancedExternalizer<ClientAddress> EXTERNALIZER = new Externalizer();
    private final Address localAddress;
 
    public ClientAddress(Address localAddress) {
       this.localAddress = localAddress;
    }
 
+   @ProtoFactory
+   ClientAddress(WrappedMessage wrappedLocalAddress) {
+      this.localAddress = WrappedMessages.unwrap(wrappedLocalAddress);
+   }
+
+   @ProtoField(number = 1, name = "localAddress")
+   WrappedMessage getWrappedLocalAddress() {
```

Comment:
99% sure it can be cast to JGroupsAddress.

Reply:
Making this change causes `TxReaperAndRecoveryTest` to fail as it relies on `PersistentUUID#randomUUID`, so I have updated the code to:

```java
   private static Address newAddress() {
      //test address isn't serializable and we just need an address that doesn't belong to the cluster (simulates a leaver)
      return new JGroupsAddress(org.jgroups.util.UUID.randomUUID());
   }
```